File: relationship-manager/admin-service/src/main/java/com/bank/rm/admin/controller/AdminController.java
package com.bank.rm.admin.controller;

import com.bank.rm.common.dto.ApiResponse;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import java.util.Map;

@RestController
@RequestMapping("/api/v1/admin")
public class AdminController {

    @GetMapping("/health")
    public ResponseEntity<ApiResponse<Map<String, String>>> healthCheck() {
        return ResponseEntity.ok(ApiResponse.ok(Map.of(
            "service", "admin-service",
            "status", "UP",
            "version", "1.0.0"
        )));
    }

    @GetMapping("/services")
    public ResponseEntity<ApiResponse<Map<String, Object>>> getServiceStatus() {
        return ResponseEntity.ok(ApiResponse.ok(Map.of(
            "auth-service", Map.of("port", 8081, "status", "configured"),
            "customer-profile-service", Map.of("port", 8082, "status", "configured"),
            "conversation-service", Map.of("port", 8083, "status", "configured"),
            "risk-profiling-service", Map.of("port", 8084, "status", "configured"),
            "product-catalog-service", Map.of("port", 8085, "status", "configured"),
            "recommendation-service", Map.of("port", 8086, "status", "configured"),
            "wealth-projection-service", Map.of("port", 8087, "status", "configured"),
            "followup-orchestrator", Map.of("port", 8088, "status", "configured"),
            "notification-service", Map.of("port", 8089, "status", "configured"),
            "analytics-service", Map.of("port", 8090, "status", "configured"),
            "admin-service", Map.of("port", 8091, "status", "configured"),
            "reminder-service", Map.of("port", 8092, "status", "configured"),
            "api-gateway", Map.of("port", 8080, "status", "configured")
        )));
    }
}

File: relationship-manager/admin-service/src/main/kotlin/com/bank/rm/admin/controller/AdminController.kt
package com.bank.rm.admin.controller

import com.bank.rm.common.dto.ApiResponse
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1/admin")
class AdminController {

    @GetMapping("/health")
    fun health(): ApiResponse<Map<String, String>> =
        ApiResponse(success = true, data = mapOf(
            "status" to "UP",
            "service" to "admin-service",
            "version" to "1.0.0"
        ))

    @GetMapping("/config/conversation-templates")
    fun getConversationTemplates(): ApiResponse<Map<String, Any>> =
        ApiResponse(success = true, data = mapOf(
            "phases" to listOf("GREETING", "PERSONAL", "FINANCIAL", "GOALS", "RISK_ASSESSMENT",
                "RECOMMENDATION", "CHANNEL_PREF", "FOLLOWUP_SCHEDULE", "COMPLETED"),
            "aiModel" to "gpt-4o",
            "maxConversationLength" to 100,
            "sessionTimeoutMinutes" to 30
        ))

    @GetMapping("/config/risk-model")
    fun getRiskModelConfig(): ApiResponse<Map<String, Any>> =
        ApiResponse(success = true, data = mapOf(
            "model" to "xgboost-v1.0",
            "features" to 15,
            "riskCategories" to listOf("CONSERVATIVE", "MODERATE", "AGGRESSIVE", "VERY_AGGRESSIVE"),
            "refreshIntervalDays" to 90,
            "shapEnabled" to true
        ))

    @GetMapping("/config/notification-channels")
    fun getNotificationConfig(): ApiResponse<Map<String, Any>> =
        ApiResponse(success = true, data = mapOf(
            "channels" to mapOf(
                "SMS" to mapOf("provider" to "Twilio", "enabled" to true),
                "EMAIL" to mapOf("provider" to "SendGrid", "enabled" to true),
                "WHATSAPP" to mapOf("provider" to "Twilio", "enabled" to true),
                "PHONE" to mapOf("provider" to "Manual Queue", "enabled" to true),
                "PUSH" to mapOf("provider" to "FCM", "enabled" to false)
            )
        ))
}
